modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,46 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
-
 				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
+
+					if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
+						continue;
+					}
+
+					var sizes = media[i].media_details.sizes;
+					var sizes_keys = Object.keys( sizes );
 					var srcset = images[j].getAttribute( 'srcset' );
-					if ( srcset ) {
-						images[j].setAttribute( 'srcset', srcset.replace( /\.webp(\s)/ig, ext[0] + '$1' ) );
+
+					// If a full image is present in srcset, it should be updated.
```

Comment:
if the full image is the img src url and there is no srcset we should be done here

Reply:
Thank you for the review. Can you please share more information about this?

- Currently, we update IMG src after the sizes checking because if we update image src then we can't replace the full image src in srcset.
```<img data-dominant-color="5b5b5b" data-has-transparency="false" style="--dominant-color: #5b5b5b;" width="1080" height="608" data-id="2105" src="http://localhost/wordpress/wp-content/uploads/2022/07/car-jpeg.webp" alt="" class="not-transparent wp-image-2105" srcset="http://localhost/wordpress/wp-content/uploads/2022/07/car-jpeg.webp 1080w, http://localhost/wordpress/wp-content/uploads/2022/07/car-300x169-jpeg.webp 300w, http://localhost/wordpress/wp-content/uploads/2022/07/car-1024x576-jpeg.webp 1024w, http://localhost/wordpress/wp-content/uploads/2022/07/car-768x432-jpeg.webp 768w" sizes="(max-width: 1080px) 100vw, 1080px">```

Reply:
I think what @adamsilverstein is saying is we could check if the `src` is the `full` size image and no `srcset` attribute exists, we don't even need the loop below.

However, please see my other comment on the inconsistency in our REST API implementation. `full` is treated just like any other image size by the REST API endpoint for attachments in core, so we should do that too. This then also means we shouldn't have any logic that is "special" for the full size anymore in this JS code, since it will be like any other size.

Reply:
Thank you for more information, Felix

This feedback is addressed with the REST API endpoint changes.